app/ts/utils/addressBookTypes.ts

```diff
@@ -0,0 +1,106 @@
+import * as funtypes from 'funtypes'
+import { EthereumAddress, EthereumQuantity, LiteralConverterParserFactory } from './wire-types.js'
+
+export type EntrySource = funtypes.Static<typeof EntrySource>
+export const EntrySource = funtypes.Union(
+	funtypes.Literal('DarkFloristMetadata'),
+	funtypes.Literal('User'),
+	funtypes.Literal('Interceptor'),
+	funtypes.Literal('OnChain'),
+	funtypes.Literal('FilledIn'),
+)
+
+export type AddressInfo = funtypes.Static<typeof AddressInfo>
+export const AddressInfo = funtypes.ReadonlyObject({
+	name: funtypes.String,
+	address: EthereumAddress,
+	askForAddressAccess: funtypes.Union(funtypes.Boolean, funtypes.Literal(undefined).withParser(LiteralConverterParserFactory(undefined, true))),
```

Comment:
can also use `funtypes.Undefined`

Reply:
that would not convert the undefined to true. Or what do you mean?